src/context/WalletProvider/WalletProvider.tsx

```diff
@@ -515,6 +531,73 @@ export const WalletProvider = ({ children }: { children: React.ReactNode }): JSX
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [state.adapters, state.keyring])
 
+  const resetState = useCallback(() => dispatch({ type: WalletActions.RESET_STATE }), [])
+
+  // Register a MetaMask-like (EIP-1193) provider on wallet connect or load
+  const onProviderChange = useCallback(
+    async (localWalletType: KeyManager | null) => {
+      if (!localWalletType) return
+      setWalletType(localWalletType)
```

Comment:
Just confirming: if we don't have a `walletType` and early exit, we are cool that `walletType` got set?

Reply:
That's on purpose yeah, to avoid duplicate runs. It will then be set, and the next run will go through - not setting it if we don't have a `walletType` also allows us to get proper `handleAccountsOrChainChanged` invalidation/early return in the follow-up PRs